src/webgl/p5.RendererGL.Immediate.js

```diff
@@ -49,6 +49,11 @@ p5.RendererGL.prototype.beginShape = function(mode) {
     this.immediateMode.lineVertexBuffer = this.GL.createBuffer();
     this.immediateMode.lineNormalBuffer = this.GL.createBuffer();
     this.immediateMode.pointVertexBuffer = this.GL.createBuffer();
+    this.immediateMode._bezierVertex = [];
+    this.immediateMode._quadraticVertex = [];
+    this.immediateMode._curveVertex_x = [];
+    this.immediateMode._curveVertex_y = [];
+    this.immediateMode._curveVertex_z = [];
```

Comment:
Why does curveVertex store x ,y ,& z separately?

Reply:
Yes I had that in my mind as well. But there are some problems using a single array.
1) I would have to give proper indices when I'm converting x,y and z coordinates using Bezier2Catmull(). This would require a loop, or passing 4 vertices with their proper vertices.

2) I would have to use splice now when I have to shift the array. splice(0,2) for 2D curves and splice(0,3) for 3D curves. This is a lot slower than shift(). See [this](https://stackoverflow.com/questions/10742369/javascript-shift-versus-splice-are-these-statements-equal).